trustmanager/x509utils.go

```diff
@@ -507,18 +505,18 @@ func X509PublicKeyID(certPubKey data.PublicKey) (string, error) {
 	if err != nil {
 		return "", err
 	}
-	var finalAlgorithm data.KeyAlgorithm
+	pubKeyBytes, err := x509.MarshalPKIXPublicKey(cert.PublicKey)
+	if err != nil {
+		return "", err
+	}
 
+	var key data.PublicKey
 	switch certPubKey.Algorithm() {
 	case data.ECDSAx509Key:
-		finalAlgorithm = data.ECDSAKey
+		key = data.NewECDSAPublicKey(pubKeyBytes)
```

Comment:
Not related to your changes, but it feels like that we blindly create a new key with `pubKeyBytes`, independently of what `certPubKey.Algorithm()` was.

What I mean is: it might be possible to pass an RSA key, with a the Algorithm of `data.ECDSAx509Key`, and this method would still succeed in returning an ID, correct?